app/src/main/java/com/lagradost/cloudstream3/ui/library/LibraryFragment.kt

```diff
@@ -163,7 +164,16 @@ class LibraryFragment : Fragment() {
                     return true
                 }
 
-                libraryViewModel.sort(ListSorting.Query, newText)
+                val callback = Runnable {
+                    libraryViewModel.sort(ListSorting.Query, newText)
+                }
+
+                binding?.mainSearch?.removeCallbacks(callback)
```

Comment:
While this seams like it should work it does not in fact do because callback is a local variable and not scoped properly it removes nothing making this pr broken. 

If you want to fix this then put the callback outside the scope of setOnQueryTextListener and do something like 

```kotlin
val searchCallback = Runnable {
    val newText = binding?.mainSearch?.query?.toString() ?: return@Runnable
    libraryViewModel.sort(ListSorting.Query, newText)
}
```

Reply:
Right now it queries for every character, as removeCallbacks does not remove the callback